pkg/bmc/redfish.go

```diff
@@ -11,30 +11,45 @@ func init() {
 	registerFactory("redfish+https", newRedfishAccessDetails)
 	registerFactory("redfish-virtualmedia", newRedfishVirtualMediaAccessDetails)
 	registerFactory("ilo5-virtualmedia", newRedfishVirtualMediaAccessDetails)
+	registerFactory("idrac-virtualmedia", newiDracRedfishVirtualMediaAccessDetails)
 }
 
-func newRedfishAccessDetails(parsedURL *url.URL) (AccessDetails, error) {
+func redfishDetails(parsedURL *url.URL) *redfishAccessDetails {
 	return &redfishAccessDetails{
 		bmcType: parsedURL.Scheme,
 		host:    parsedURL.Host,
 		path:    parsedURL.Path,
-	}, nil
+	}
+}
+
+func newRedfishAccessDetails(parsedURL *url.URL) (AccessDetails, error) {
+	return redfishDetails(parsedURL), nil
 }
 
 func newRedfishVirtualMediaAccessDetails(parsedURL *url.URL) (AccessDetails, error) {
-	return &redfishAccessDetails{
-		bmcType:        parsedURL.Scheme,
-		host:           parsedURL.Host,
-		path:           parsedURL.Path,
-		isVirtualMedia: true,
+	return &redfishVirtualMediaAccessDetails{
+		*redfishDetails(parsedURL),
+	}, nil
+}
+
+func newiDracRedfishVirtualMediaAccessDetails(parsedURL *url.URL) (AccessDetails, error) {
+	return &redfishiDracVirtualMediaAccessDetails{
```

Comment:
It's slightly jarring that redfish and iDrac are reversed in the typename vs. the function name

Reply:
I mean, what do you expect from a language without a real type system? :duck:  :duck:

Reply:
LOL, I just read that again. My bad.